### ISSUE TITLE
Annotations on HTML elements generated for a question

### ISSUE THREAD

**Issue by abchugh** [KWARC/sTeX#378]
For documents containing questions, we need annotations that identify different parts of the problem structure.
Example document: https://gl.mathhub.info/problems/IWGS/-/blob/main/source/progintro/quiz/algorithm.de.tex

The top-level div already has `property` set to `stex:problem`. I will continue to use this to locate the outermost question div.

The following should be marked with appropriate 'property' value:
1. question title
2. question text 
3. type of question (single/multi select or text-input)
4. Parent div of each option
5. Answer option text (that should be shown to the user)
6. Whether the option should have been selected
7. The div that should be shown as feedback after the answer is provided

**Comment by kohlhase** [comment]
Only Dennis can do this.

**Comment by Jazzpirate** [comment]
1. Question title: (Often) doesn't exist (but arguably should)
2. Question text: Is equivalent to the sproblem-environment itself
3. Type of Question: doesn't exist (multiple choice blocks / text-input can be freely mixed anywhere. Not sure if it should be that way, but it makes sense to me)
4. I don't know what that means...
5. exists in sTeX3.3
6. exists in sTeX3.3
7. exists in sTeX3.3

...in general I can only markup things that are explicitly "marked up" in the latex sources. If we want a "type of question", we have to provide that in the {sproblem}-environment. If we want titles, we need to provide a title-argument to the sproblem-environment.

**Comment by abchugh** [comment]
One thing that missing in current annotations: For a multiple-choice question, should the user select one option or multiple options?

I don't think sTeX even supports specifying this info but I have seen questions from Michael where there are multiple correct options.

But in most scenarios, there is one correct option.

**Comment by kohlhase** [comment]
You are completely right, the reason for this is that I do not really understand the issues involved. I will read up on this and make some sTeX suggestions (@abchugh, do you have any?) and see how this goes into more annotations in rustex with @Jazzpirate.

**Comment by Jazzpirate** [comment]
I would guess a binary optionaly argument to the `mcb` environment would suffice...?
